### ISSUE TITLE
Recompile for modern MacOS

### ISSUE THREAD

**Issue by thegushi** [david-schmidt/gsport#8]
Hey there,

Would it be possible to have a compiled version for modern MacOS machines, or are there major code changes required?

**Comment by david-schmidt** [comment]
It is possible, and there are major code changes required.  We have a PR that does the SDL work (https://github.com/david-schmidt/gsport/pull/5) but it has repercussions for backwards compatibility on older platforms, so it hasn't been adopted yet.

**Comment by thegushi** [comment]
On Mon, 15 Jun 2020, David Schmidt wrote:

> 
> It is possible, and there are major code changes required. We have a PR that does the SDL work (#5) but it has repercussions for backwards compatibility on older
> platforms, so it hasn't been adopted yet.

Hey there,

I've also looked into building it under HomeBrew (brew.sh), and was 
surprised there wasn't already a cask for it.

Would this be an option moving forward, or is the license somehow 
incompatible or something like that?

-Dan

> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or
> unsubscribe.[AAIWKKBYHZ7LGCGP374KJQ3RWYUVFA5CNFSM4N43CTG2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEZSSQBI.gif]
> 
> 
>

-- 

"I hate Windows"

-Tigerwolf, Anthrocon 2004

--------Dan Mahoney--------
Techie,  Sysadmin,  WebGeek
Gushi on efnet/undernet IRC
FB:  fb.com/DanielMahoneyIV
LI:   linkedin.com/in/gushi
Site:  http://www.gushi.org
---------------------------

**Comment by david-schmidt** [comment]
No, it's just a matter of deciding how to proceed.  Older systems don't support SDL, and need the older graphics rendering system.  Newer systems support SDL now and probably some time into the future, so it's a good path going forward.  It's just that it'll require two independent branches that require a little different concept of development.  Not hard, just not a big priority for anyone that has the time to do the work.

**Comment by NJRoadfan** [comment]
A 64-bit port (and future ARM) would require either a native Cocoa driver or moving to SDL (which handles the Cocoa part behind the scenes).

**Comment by aramis-erak** [comment]
Renewing the request, as no intel macs have shipped in several years now, and I'd love to be able to run this on Tahoe 26.3... but it's codebase is too old for Tahoe's Intel emulation.
Kent's not got printer emulation...
